src/npc_attack.cpp

```diff
@@ -172,40 +214,63 @@ npc_attack_rating npc_attack_melee::evaluate_critter( const npc &source,
     if( !critter ) {
         return npc_attack_rating{};
     }
-    const int distance_to_me = rl_dist( source.pos(), critter->pos() );
+
     const double damage{ weapon.effective_dps( source, *critter ) };
-    const Creature::Attitude att = source.attitude_to( *critter );
-    double potential = damage * npc_attack_constants::attitude_multiplier.at( att ) -
-                       ( distance_to_me - 1 );
+    const int reach_range{ weapon.reach_range( source ) };
+    const int distance_to_me = clamp( rl_dist( source.pos(), critter->pos() ) - reach_range, 0, 10 );
+    // Multiplier of 0.5f to 1.5f based on distance
+    const float distance_multiplier = 1.5f - distance_to_me * 0.1f;
+    double potential = damage * distance_multiplier;
+
     if( damage >= critter->get_hp() ) {
         potential *= npc_attack_constants::kill_modifier;
     }
     if( target && target->pos() == critter->pos() ) {
         potential *= npc_attack_constants::target_modifier;
     }
+
     return npc_attack_rating( std::round( potential ), critter->pos() );
 }
 
 void npc_attack_gun::use( npc &source, const tripoint &location ) const
 {
-    const item &weapon = *gunmode;
-    if( !weapon.ammo_sufficient( &source ) ) {
-        source.do_reload( weapon );
+    if( !source.is_wielding( gun ) ) {
+        if( !source.wield( gun ) ) {
+            debugmsg( "ERROR: npc tried to equip a weapon it couldn't wield" );
+        }
+        return;
+    }
+
+    if( !gun.ammo_sufficient() ) {
```

Comment:
```suggestion
    if( !gun.ammo_sufficient( &source ) ) {
```

Reply:
Thanks, I messed up when fixing the conflicts.